### PR TITLE
Fixed ABORT_ON_WASM_EXCEPTIONS to work with the recent dynCall changes

### PR DESCRIPTION
**Pull request by tklajnscek**
Fixed `ABORT_ON_WASM_EXCEPTIONS` to work with the recent dynCall changes where functions can be called via the WASM table directly, bypassing WASM exports which broke the instrumentation for embind calls.

This is done by overriding `wasmTable.get` on the JS side. Open to better suggestions, but this one seemed the most analogous the way the instrumentation for `exports` was done.

Also made the test for this actually test multiple ways of calling a function (raw, cwrap, ccall, embind) which would've caught this.

Finally, fixed the 2.0.3 change log entry to reflect the correct setting name (i.e. ABORT_ON_WASM_EXCEPTIONS not ABORT_ON_EXCEPTIONS). Not sure if this is legal to do, but for anyone reading it it would help if the name was correct :flushed:

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -745,49 +745,72 @@ var abortWrapperDepth = 0;
...
+  // A cache for wrappers based on the original function reference so we don't end up
+  // creating the same wrappers over and over again
+  var wrapperCache = {};
+
+  // Override the wasmTable get function to return the wrappers
+  var realGet = wasmTable.get; 
+  wasmTable.get = function(i) { 
+    var original = realGet.call(wasmTable, i);
+    var wrapper = wrapperCache[original];
+    if(!wrapper) {
```
Comment (sbc100) [inline src/preamble.js:792]: Space after `if` keyword here and below.

File: src/preamble.js [hunk 2]
```diff
@@ -745,49 +745,72 @@ var abortWrapperDepth = 0;
...
+  // Override the exported functions with the wrappers and copy over any other symbols
+  var instExports = {};
+  for (var name in exports) {
+      var original = exports[name];
+      if (typeof original === 'function') {
+        var wrapper = wrapperCache[original];
+        if(!wrapper) {
+          wrapper = makeWrapper(original);
+          wrapperCache[original] = wrapper;
+        }
```
Comment (sbc100) [inline src/preamble.js:808]: Avoid duplicating these 4 lines by having the cache built into makeWrapper itself?  Just have something like `if (isCached) return cachedVersion` at the top of makeWrapper?

File: tests/core/test_abort_on_exception.out [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
 again
-true
\ No newline at end of file
+program has already aborted!
+ccall
+program has already aborted!
+cwrap
+program has already aborted!
+embind
+program has already aborted!
```
Comment (sbc100) [inline tests/core/test_abort_on_exception.out:10]: add newline

**Review by sbc100 — APPROVED** [review]
lgtm % comments

File: src/preamble.js (hunk 2 above)
Comment (tklajnscek) [reply src/preamble.js:808]: dOh! How obvious :) Thanks!

**Comment by tklajnscek** [comment]
I'll hold on merging while I look at https://github.com/emscripten-core/emscripten/issues/12277

**Review by sbc100 — APPROVED** [review]
lgtm.. let me know if you are ready for this to land.

**Comment by tklajnscek** [comment]
Please go ahead and merge this and I'll look at that issue separately on Monday. Thanks!

**Pull request merged**